exercises/concept/poetry-club-door-policy/door-policy.spec.js

```diff
@@ -1,245 +1,142 @@
-  get acrostic() {
-    return this.lines.map((line) => [...line].shift()).join('');
-  }
-
-  get telestich() {
-    return this.lines.map((line) => [...line.trim()].pop()).join('');
-  }
-
-  *[Symbol.iterator]() {
-    yield* this.lines[Symbol.iterator]();
-  }
-}
-
-class FrontDoorGuard {
-  /**
-   * @param {Poem} poem
-   */
-  constructor(poem) {
-    this.poem = poem;
-  }
-
-  /**
-   * Recites a poem at the door, asking the patron for a response
-   *
-   * @returns {string} key
-   */
-  recite() {
-    const patron = frontDoorPatron;
-    const key = [];
-
-    for (const line of this.poem) {
-      key.push(patron(line));
-    }
-
-    return key.join('');
-  }
-
-  assert() {
-    const password = frontDoorPassword(this.recite());
-
-    expect(password).toBe(stringify(generateCapitalized(this.poem.acrostic)));
-  }
-}
-
-class BackDoorGuard {
-  constructor(poem) {
-    this.poem = poem;
-  }
-
-  /**
-   * Recites a poem at the door, asking the patron for a response
-   *
-   * @returns {string} key
-   */
-  recite() {
-    const patron = backDoorPatron;
-    const key = [];
-
-    for (const line of this.poem) {
-      key.push(patron(line));
-    }
-
-    return key.join('');
-  }
-
-  assert() {
-    const password = backDoorPassword(this.recite());
-
-    expect(password).toBe(
-      stringify(generateCapitalized(this.poem.telestich)).concat(', please')
+const recite = (poem, responseFn) => {
+  return poem.map((line) => responseFn(line));
+};
+
+//Shire Horse, by Michael Lockwood
+const SHIRE_HORSE = [
+  'Stands so high',
+  'Huge hooves too',
+  'Impatiently waits for',
+  'Reins and harness',
+  'Eager to leave',
+];
+
+//Shire Horse, by Michael Lockwood
+//with leading and trailing whitespace
+const SHIRE_HORSE_WITH_SPACES = [
+  '  Stands so high',
+  '  Huge hooves too  ',
+  '    Impatiently waits for',
+  ' Reins and harness   ',
+  '         Eager to leave  ',
+];
+
+//Summer, by John Albert Caballero
+const SUMMER = [
+  'Sunshine warming my toes,',
+  'Underwater fun with my friends.',
+  'Making homemade ice cream on the porch,',
+  'Many long nights catching fireflies.',
+  'Early morning walks to the creek,',
+  'Reveling in the freedom of lazy days.',
+];
+
+//Sophia, by John Albert Caballero
+const SOPHIA = [
+  'Serene, calming quality',
+  'Organized, you always have it together',
+  'Picturesque, strikingly beautiful',
+  'Honest, so genuine',
+  'Imaginative, a creative mind',
+  'Alluring, so attractive',
+];
+
+//Code Work, by Derk-Jan Karrenbeld
+const CODE_WORK = [
+  'Compilers intensily bestow',
+  'On commencing without ego',
+  'Different processes ajar',
+  'Exit with zero quick',
+];
+
+describe('front door response', () => {
+  test('should output a character per line', () => {
+    const expectedLetters = recite(SHIRE_HORSE, frontDoorResponse);
+    expect(expectedLetters.length).toBe(SHIRE_HORSE.length);
+  });
+
+  xtest('should take the first character of each line', () => {
```

Comment:
For concept exercises all tests should be unskipped as they are not meant to be done TDD style.